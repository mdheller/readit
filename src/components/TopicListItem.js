import React, { Component } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import PropTypes from 'prop-types';

/**
 * The TopicListItem component defines the layout for a single topic in
 * TopicList.
 */
class TopicListItem extends Component {
  render() {
    const {
      topic: {
        id, title, score
      },
      onUpPressed,
      onDownPressed,
    } = this.props;
    return (
      <View style={styles.cell}>
        <View style={styles.container}>
          <Text style={styles.title} >{title}</Text>
          <View style={styles.details}>
            <Text>{score} points</Text>
            <View style={styles.flexSpacing} />
            <Button
              title='Up'
              onPress={() => onUpPressed()}
            />
            <View style={styles.buttonSpacing} />
            <Button
              title='Dn'
              onPress={() => onDownPressed()}
            />
          </View>
        </View>
        <View style={styles.divider} />
      </View>
    );
  }
}

TopicListItem.propTypes = {
  topic: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
  }),
  onUpPressed: PropTypes.func.isRequired,
  onDownPressed: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  cell: {
    flexDirection: 'column',
  },
  container: {
    flexDirection: 'column',
    paddingVertical: 16,
    paddingHorizontal: 16,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 18,
    paddingBottom: 8,
  },
  details: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  flexSpacing: {
    flex: 1,
  },
  buttonSpacing: {
    width: 8,
  },
  divider: {
    height: 1,
    backgroundColor: '#a9a9a944',
  },
});

export default TopicListItem;
