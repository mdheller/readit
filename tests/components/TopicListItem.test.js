import 'react-native';
import React from 'react';
import TopicListItem from '../../src/components/TopicListItem';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <TopicListItem
      topic={{ id: 1, title: 'test', score: 1 }}
      onUpPressed={() => {}}
      onDownPressed={() => {}}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
