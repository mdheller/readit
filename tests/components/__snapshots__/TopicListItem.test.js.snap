// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "paddingHorizontal": 16,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
          "paddingBottom": 8,
        }
      }
    >
      test
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        1
         points
      </Text>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Up
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 8,
          }
        }
      />
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Dn
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#a9a9a944",
        "height": 1,
      }
    }
  />
</View>
`;
